### PR TITLE
Test ManyToMany with multiple schemas

### PR DESCRIPTION
**Pull request by jenovateurs**
Hello team!
I created a test for the PR #14716 .
I added new table in the schema.sql.
I added new models with different schema.
I fix an error in .env.exemple of nanobox.

* Type: code quality
* Link to issue: #14716

**In raising this pull request, I confirm the following:**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)
- [x] I have checked that another pull request for this purpose does not exist
- [x] I wrote some tests for this PR
- [ ] I have updated the relevant CHANGELOG
- [ ] I have created a PR for the [documentation](https://github.com/phalcon/docs) about this change

Thanks

### CODE REVIEW

**Comment by ruudboon** [comment]
@jenovateurs Thank you! I cancelled Appveyor to give priority to the release. I will restart it afterwards

**Comment by jenovateurs** [comment]
@ruudboon No problem, thanks for the info ;-).

**Comment by Jeckerson** [comment]
@jenovateurs Use fixtures migration classes instead of raw SQL.

Example here: `tests/_data/fixtures/Migrations/InvoicesMigration.php`

**Comment by jenovateurs** [comment]
@Jeckerson No problem about fixtures migration. Should I delete my raw SQL in the schemas files ?

**Comment by ruudboon** [comment]
@jenovateurs yes, you can delete the sql files you've added. We will move everything to the migrations soon. After that we can fully clean up.

**Comment by jenovateurs** [comment]
What can I do about that: 

```
In Db.php line 278:
                                                                               
  Db: SQLSTATE[42000]: Syntax error or access violation: 1044 Access denied f  
  or user 'phalcon'@'%' to database 'hardware'                                 
  SQL query being executed:                                                    
  CREATE SCHEMA IF NOT EXISTS hardware  
```

**Comment by Jeckerson** [comment]
@jenovateurs Did you rebased with 4.0.x?

**Comment by jenovateurs** [comment]
@Jeckerson I've just rebased now but I still have the same issue.

**Comment by ruudboon** [comment]
@jenovateurs I can have a look at it tomorrow. Must be something related to permissions on the github actions.

**Comment by jenovateurs** [comment]
@ruudboon I think the problem is here : 
https://github.com/phalcon/cphalcon/blob/4.0.x/tests/_envs/pgsql.yml
In the database, I haven't got the permission to create a new schema.
I can create the new schema in the file environnement or add more privileges. 
We will see tomorrow, no problem ;-).

**Comment by niden** [comment]
@jenovateurs If you have a better understanding and knowledge about Postgres make any necessary changes that are needed. 

Other than the area that you pointed regarding the schema creation, I think the rest of the database tests could run on postgres without major hitches (provided that the migrations are created)

**Comment by ruudboon** [comment]
@jenovateurs Probably your latest commit will fail because you don't have permissions. 
To fix this we have a few options.
- Run tests as root
- Initialise permissions on starting of mysql/postgres containers.

Besides that I would like to use schema's that will be used for the full suite. This could be `hardware` but maybe we could go for something more general like `app` or `private` (we already have the `public` I think)

**Comment by ruudboon** [comment]
Let’s run the suite as root for now. Looks like we need to extend the mysql and postgres containers otherwise. 
Can you change the example env and actions env to use the root user?

File: .github/workflows/codecoverage.yml [hunk 1]
```diff
@@ -166,6 +166,7 @@ jobs:
           DATA_POSTGRES_PORT: ${{ job.services.postgres.ports['5432'] }}
           DATA_REDIS_PORT: ${{ job.services.redis.ports['6379'] }}
           DATA_MEMCACHED_PORT: ${{ job.services.memcached.ports['11211'] }}
+	  DATA_MYSQL_USER: root
```
Comment (ruudboon) [inline .github/workflows/codecoverage.yml]: Can indent this?

File: tests/_data/assets/db/schemas/mysql_schema.sql [hunk 2]
```diff
@@ -12792,4 +12792,4 @@ CREATE TABLE `tiny_int_test` (
   `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
   `test` tinyint(1) unsigned DEFAULT NULL,
   PRIMARY KEY (`id`)
-) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
```
Comment (ruudboon) [inline tests/_data/assets/db/schemas/mysql_schema.sql]: Please end files with a new line

File: tests/_data/assets/db/schemas/postgresql_schema_nanobox.sql [hunk 3]
```diff
@@ -7045,4 +7045,4 @@ GRANT ALL ON SCHEMA public TO PUBLIC;
 
 --
 -- PostgreSQL database dump complete
---
+--
```
Comment (ruudboon) [inline tests/_data/assets/db/schemas/postgresql_schema_nanobox.sql]: same

**Review by ruudboon — CHANGES_REQUESTED** [review]

File: .github/workflows/codecoverage.yml [hunk 4]
```diff
@@ -166,7 +166,7 @@ jobs:
           DATA_POSTGRES_PORT: ${{ job.services.postgres.ports['5432'] }}
           DATA_REDIS_PORT: ${{ job.services.redis.ports['6379'] }}
           DATA_MEMCACHED_PORT: ${{ job.services.memcached.ports['11211'] }}
-	  DATA_MYSQL_USER: root
+	        DATA_MYSQL_USER: root
```
Comment (ruudboon) [inline .github/workflows/codecoverage.yml]: I think there is a tab here. Please use spaces only.

**Review by ruudboon — DISMISSED** [review]

**Comment by jenovateurs** [comment]
It's the sqlite test that trigger the error. Because I haven't the SQL script. I don't know if it's possible to create a schema with sqlite.

**Comment by jenovateurs** [comment]
I skiped sqlite for this special test so if you have another idea... I'm all ears (or eyes) :-).

**Review by ruudboon — APPROVED** [review]

**Comment by niden** [comment]
I wish I had seen this earlier. Nothing you did Jeremy but the goal is to move away from the Robots and related tables and use a more realistic schema such as invoices/customers/products etc. Also by doing this we ensure that we use only models we need and not throw one model/table here just for one test.

Great work nonetheless. Thank you for taking the initiative on this.

Also for the future if you do not want a test to run for a particular RDBMs you can use the `getDriver()` method from the database helper.

**Pull request merged**